### PR TITLE
Implement unhealthy state and transitions

### PR DESCRIPTION
**Pull request by kriskowal**
Unit test peer state transitions

### CODE REVIEW

**Comment by kriskowal** [comment]
r @jcorbin

File: node/peer_states.js [hunk 1]
```diff
@@ -22,13 +22,65 @@
...
+ * - TODO
+ *
+ * They also inherit:
+ * - channel.timers
+ * - channel.random
+ */
+
+var symptoms = {
+    'BadRequest': false, // not an indicator of bad health
+    'Cancelled': false, // not an indicator of bad health
```
Comment (breerly) [inline node/peer_states.js]: Are there not constants that can be used in place of these strings?

Comment (kriskowal) [inline node/peer_states.js]: There are, but as it happens there is nothing to be gained by a level of indirection through that lookup table. We are getting the code as a string from the error classifier.

Comment (breerly) [inline node/peer_states.js]: Aaah, makes sense ;)

**Comment by kriskowal** [comment]
Looks like some commits got redrawn in the rebase. Fixing.

File: node/test/send.js [hunk 2]
```diff
@@ -203,57 +206,64 @@ allocCluster.test('request().send() to a pool of servers', 4, function t(cluster
             res.sendOk(arg2, arg3 + ' served by ' + chanNum);
         });
         channel.peers.add(chan.hostPort);
+        var peer = channel.peers.get(chan.hostPort);
+        var conn = peer.connect(chan.hostPort);
+        console.log(chan.hostPort);
```
Comment (jcorbin) [inline node/test/send.js]: spam

File: node/test/send.js [hunk 3]
```diff
@@ -203,57 +206,64 @@ allocCluster.test('request().send() to a pool of servers', 4, function t(cluster
...
+        conn.on('identified', ready.signal);
```
Comment (jcorbin) [inline node/test/send.js]: this whole pattern might be worthy of getting hauled back into allocCluster or an ancillary utility function around it.

**Comment by jcorbin** [comment]
lgtm, spammy test is spammy

**Pull request merged**